diff --git a/arch/arm/dts/rk3588s-orangepi-5.dts b/arch/arm/dts/rk3588s-orangepi-5.dts
index 8f399c4317..3997984e42 100644
--- a/arch/arm/dts/rk3588s-orangepi-5.dts
+++ b/arch/arm/dts/rk3588s-orangepi-5.dts
@@ -238,7 +238,7 @@
 &pcie2x1l2 {
 	reset-gpios = <&gpio3 RK_PD1 GPIO_ACTIVE_HIGH>;
 	vpcie3v3-supply = <&vcc3v3_pcie20>;
-	status = "okay";
+	status = "disabled";
 };
 
 &pinctrl {
@@ -660,3 +660,7 @@
 &usb_host1_ohci {
 	status = "okay";
 };
+
+&sata0 {
+	status = "okay";
+};
diff --git a/arch/arm/dts/rk3588s.dtsi b/arch/arm/dts/rk3588s.dtsi
index 61a9a11c3b..05028012ce 100644
--- a/arch/arm/dts/rk3588s.dtsi
+++ b/arch/arm/dts/rk3588s.dtsi
@@ -1390,6 +1390,8 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 		status = "disabled";
+		phys = <&combphy0_ps PHY_TYPE_SATA>;
+		phy-names = "sata-phy";
 
 		sata-port@0 {
 			reg = <0>;
