diff --git a/stagit-index.c b/stagit-index.c
index 6402296bd1c0..c5cd9e4d16e8 100644
--- a/stagit-index.c
+++ b/stagit-index.c
@@ -151,7 +151,7 @@ writelog(FILE *fp)
 
 	fputs("<tr><td><a href=\"", fp);
 	percentencode(fp, stripped_name, strlen(stripped_name));
-	fputs("/log.html\">", fp);
+	fputs("/\">", fp);
 	xmlencode(fp, stripped_name, strlen(stripped_name));
 	fputs("</a></td><td>", fp);
 	xmlencode(fp, description, strlen(description));
diff --git a/stagit.c b/stagit.c
index 409714b728ef..5e8d8489748d 100644
--- a/stagit.c
+++ b/stagit.c
@@ -947,7 +947,7 @@ writeatom(FILE *fp, int all)
 }
 
 size_t
-writeblob(git_object *obj, const char *fpath, const char *filename, size_t filesize)
+writeblob(git_object *obj, const char *fpath, const char *filename, const char *entrypath, size_t filesize)
 {
 	char tmp[PATH_MAX] = "", *d;
 	const char *p;
@@ -971,7 +971,9 @@ writeblob(git_object *obj, const char *fpath, const char *filename, size_t files
 	writeheader(fp, filename);
 	fputs("<p> ", fp);
 	xmlencode(fp, filename, strlen(filename));
-	fprintf(fp, " (%zuB)", filesize);
+	fprintf(fp, " (%zuB) - <a href=\"%sraw/", filesize, relpath);
+	percentencode(fp, entrypath, strlen(entrypath));
+	fputs("\">Raw</a>", fp);
 	fputs("</p><hr/>", fp);
 
 	if (git_blob_is_binary((git_blob *)obj))
@@ -1070,7 +1072,7 @@ writefilestree(FILE *fp, git_tree *tree, const char *path)
 			}
 
 			filesize = git_blob_rawsize((git_blob *)obj);
-			lc = writeblob(obj, filepath, entryname, filesize);
+			lc = writeblob(obj, filepath, entryname, entrypath, filesize);
 
 			fputs("<tr><td>", fp);
 			fputs(filemode(git_tree_entry_filemode(entry)), fp);
